src/osemgrep/cli_interactive/Interactive_subcommand.ml

```diff
@@ -804,10 +817,78 @@ let stop_thread_if_turbo state =
 (* Interactive loop *)
 (*****************************************************************************)
 
+(* This thread's job is to constantly check for events (and consistently block
+   if they do not occur).
+
+   When an event is read, it will be added to a global queue, which will then
+   signal to the interactive loop that an event has occurred, and cause a
+   rerender.
+*)
+let spawn_event_thread term =
+  Thread.create
+    (fun _ ->
+      let rec loop () =
+        let e = Term.event term in
+        Queue.add e event_queue;
+        (* THINK: yield here? *)
```

Comment:
reading the terminal with Term.event will yield the thread if there's nothing to read on the terminal.